crates/cast/bin/cmd/storage.rs

```diff
@@ -63,6 +64,12 @@ pub struct StorageArgs {
 
     #[command(flatten)]
     build: CoreBuildArgs,
+
+    /// The path of the file in which the storage layout is saved. Only works if a slot
+    /// is not provided. The formatted (pretty) storage layout is shown even if this value
+    /// is provided.
+    #[arg(long, default_value = "", value_name = "output_json")]
+    pub output_json: String,
```

Comment:
we have the global `--json` flag you could use, just check `shell::is_json()` to see if cast storage was called with it. then one could just redirect output to desired file, I don't think this new arg is needed.

Reply:
Good point. The warning about fetching the source code from Etherscan is printed to `stderr` so the rest of the contents can be fetched as a formatted json from `stdout`. Compilation logs are `quiet(true)`.